File: frontend_flutter/lib/src/models/product_model.dart
import 'package:frontend_ecommerce_app/src/models/category.dart';
import 'package:frontend_ecommerce_app/src/models/user_model.dart';

class Product {
  final String id;
  final String title;
  final String description;
  final List<String> images;
  final List<String> tags;
  final int price;
  final User user;
  final Category category;

  Product({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    required this.images,
    required this.tags,
    required this.user,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      images: json['images'].cast<String>(),
      tags: json['tags'].cast<String>(),
      price: json['price'],
      user: User.fromJson(json['user']),
      category: Category.fromJson(json['category']),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "title": title,
      "description": description,
      "images": images,
      "price": price,
      "user": user.id,
      "category": category.id,
      "tags": tags,
    };
  }
}

File: server_nodejs/src/models/product.model.js
import mongoose from 'mongoose';

const ProductSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true
    },
    description: {
        type: String,
        required: false
    },
    price: {
        type: Number,
        index: true,
        required: true
    },
    category: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Category',
        required: true
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    featureImage: {
        type: String,
        required: false
    },
    images: [{
        type: String,
        required: false
    }],
    created_at: {
        type: Date,
        default: Date.now
    },
    numberLiked: {
        type: Number,
        default: 0
    },
    numberViewed: {
        type: Number,
        default: 0
    },
    updated_at: {
        type: Date,
        default: Date.now
    }
});

const Product = mongoose.model('Product', ProductSchema);

// Product.create({
//     title: 'Product 1',
//     description: 'This is product 1',
//     price: 100,
//     category: '6214544dfb5746fa1653043e',
//     user: '620b57815eb5e0b588f2f7b9',
//     image: 'https://cdn.pixabay.com/photo/2016/11/29/05/47/milk-1867468_960_720.jpg',
//     list_image: [
//         'https://cdn.pixabay.com/photo/2016/11/29/05/47/milk-1867468_960_720.jpg',
//         'https://cdn.pixabay.com/photo/2016/11/29/05/47/milk-1867468_960_720.jpg',
//         'https://cdn.pixabay.com/photo/2016/11/29/05/47/milk-1867468_960_720.jpg'

//     ],
//     created_at: Date.now(),
//     numberLiked: 0,
//     numberViewed: 0,
//     updated_at: Date.now()
// });



module.exports = Product;